argocd/Dockerfile

```
@@ -41,12 +34,12 @@ RUN curl -sSLf "https://${PACKAGE}/archive/v${ARGOCD_VERSION}.tar.gz" | \
 WORKDIR /go/src/github.com/argoproj/argo-cd
 RUN make argocd-all
 
-# Install yarn for building Web UI
-RUN apt-get update && apt-get install -y --no-install-recommends gnupg \
-    && curl -sLf https://dl.yarnpkg.com/debian/pubkey.gpg | apt-key add - \
-    && echo "deb https://dl.yarnpkg.com/debian/ stable main" | tee /etc/apt/sources.list.d/yarn.list \
-    && apt-get update && apt-get install -y --no-install-recommends yarn nodejs \
-    && rm -rf /var/lib/apt/lists/*
+# Install Node.js 12.x and yarn
+# https://github.com/nodesource/distributions/blob/master/README.md#debinstall
+RUN curl -sL https://deb.nodesource.com/setup_12.x | bash - \
```

This will evaluate the HTML message of an error if curl fails.  It's not desirable.  `-f` prevents it.
`-S` shows an error in stderr.

```suggestion
RUN curl -sSLf https://deb.nodesource.com/setup_12.x | bash - \
```